lib/srv/desktop/discovery.go

```diff
@@ -416,44 +447,29 @@ func (s *WindowsService) startDynamicReconciler(ctx context.Context) error {
 		// If we got here, the reconciler is running.
 		errCh <- nil
 
-		tickDuration := 5 * time.Minute
-		expiryDuration := tickDuration + 2*time.Minute
-
-		tick := s.cfg.Clock.NewTicker(tickDuration)
-		defer tick.Stop()
-
 		for {
 			select {
 			case desktops := <-watcher.ResourcesC:
 				start := s.cfg.Clock.Now()
 				newResources = make(map[string]types.WindowsDesktop)
 				for _, dynamicDesktop := range desktops {
+					// TODO: the reconciler's default comparison func is going to mark new resources
+					// different just because the expiry timestamp differs.
 					desktop, err := s.toWindowsDesktop(dynamicDesktop)
 					if err != nil {
 						s.cfg.Logger.WarnContext(ctx, "Can't create desktop resource", "error", err)
 						continue
 					}
-					desktop.SetExpiry(s.cfg.Clock.Now().Add(expiryDuration))
+					// TODO: can we use a long expiry here since the reconciler does explicit deletion?
+					desktop.SetExpiry(s.cfg.Clock.Now().Add(apidefaults.ServerAnnounceTTL))
+					s.cfg.Logger.WarnContext(ctx, "reconciler discovered new resource from watcher", "name", dynamicDesktop.GetName())
 					newResources[dynamicDesktop.GetName()] = desktop
 				}
 				if err := reconciler.Reconcile(ctx); err != nil {
 					s.cfg.Logger.WarnContext(ctx, "Reconciliation failed, will retry", "error", err)
 					continue
 				}
-				currentResources = newResources
-				s.cfg.Logger.DebugContext(ctx, "completed dynamic desktop reconciliation", "duration", s.cfg.Clock.Since(start), "count", len(newResources))
-			case <-tick.Chan():
```

Comment:
Now that current resources are actually backed by querying the backend/cache, I don't think we need to run the reconciler on an interval at all. If I'm not mistaken, it should be fine to only run it whenever the watcher has updated data.